### PR TITLE
{CI} Fix python version

### PR DESCRIPTION
**Pull request by qwordy**
**Description**  

CI is broken. Fix python version. An urgent hot fix PR from app service is blocked by this issue.

**Testing Guide**  

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Review by fengzhou-msft — APPROVED** [review]

**Comment by jiasli** [comment]
The title is too vague.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -92,7 +92,7 @@ jobs:
   - task: UsePythonVersion@0
     displayName: 'Use Python 3'
     inputs:
-      versionSpec: 3.x
+      versionSpec: 3.8
```
Comment (jiasli) [inline azure-pipelines.yml:95]: I previously pinned Python version for Windows (#15691). Is 3.9 causing any problem on Linux?

Comment (qwordy) [reply azure-pipelines.yml]: OK. I reverted it.

**Review by arrownj — APPROVED** [review]

File: azure-pipelines.yml [hunk 2]
```diff
@@ -122,7 +122,7 @@ jobs:
...
```
Comment (jiasli) [inline azure-pipelines.yml:125]: The `dev` branch already has this change:

https://github.com/Azure/azure-cli/blob/d9afad75dcae385e24412ed753704ac430847879/azure-pipelines.yml#L125

Comment (jiasli) [reply azure-pipelines.yml:125]: Ah. I see you are targeting the `release` branch. Remember to merge it back to `dev` later.

**Review by jiasli — APPROVED** [review]

**Pull request merged**